packages/altive_lints/lib/src/assists/wrap_with_macro_template_document_comment.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/source/source_range.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+
+/// {@template altive_lints.WrapWithMacroTemplateDocumentComment}
+/// A Dart assist that wraps an existing documentation comment with a macro
+/// template comment.
+///
+/// This assist helps in maintaining consistent documentation across the
+/// codebase by ensuring that each documentation comment is wrapped with a
+/// macro template, which can be used for generating documentation or for
+/// other tooling purposes.
+///
+/// The macro template comment follows the format:
+///
+/// ```dart
+/// /// {[@]template packageName.className}
+/// /// Existing documentation comment.
+/// /// {[@]endtemplate}
+/// ```
+///
+/// Example usage:
+///
+/// Before:
+/// ```dart
+/// /// This is a class.
+/// class MyClass {
+///   // Class implementation
+/// }
+/// ```
+///
+/// After running the assist:
+/// ```dart
+/// /// {[@]template my_package.MyClass}
+/// /// This is a class.
+/// /// {[@]endtemplate}
+/// class MyClass {
+///   // Class implementation
+/// }
+/// ```
+///
+/// {@endtemplate}
+class WrapWithMacroTemplateDocumentComment extends DartAssist {
+  /// {@macro altive_lints.WrapWithMacroTemplateDocumentComment}
+  WrapWithMacroTemplateDocumentComment();
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ChangeReporter reporter,
+    CustomLintContext context,
+    SourceRange target,
+  ) {
+    context.registry.addComment((node) {
+      if (!target.intersects(node.sourceRange)) {
+        return;
+      }
+
+      if (!node.isDocumentation) {
+        return;
+      }
```

Comment:
![badge](https://img.shields.io/badge/review-nits-inactive) 


```suggestion
      if (!target.intersects(node.sourceRange)) {
        return;
      }
      
      if (node.toSource().contains('{@template') &&
          node.toSource().contains('{@endtemplate}')) {
        return;
      }

      if (!node.isDocumentation) {
        return;
      }
```
I thought it might be worth considering adding this early return.

Reply:
Thank you!  
It certainly seems like a good idea. ✍️  

Since `toSource` was an empty string and couldn't be evaluated, I changed it to use `tokens` for evaluation!
[feat: Prevent wrapping of existing macro template documentation comments](https://github.com/altive/altive_lints/pull/78/commits/63261112e0fbd4f8b832fa8fcf1573a669d2f673)

```dart
      final currentComment = node.tokens.join();
      if (currentComment.contains('{@template') &&
          currentComment.contains('{@endtemplate}')) {
        return;
      }
```